CodeGenerator/Templates/Custom/s3/multipart_API.stencil

```diff
@@ -0,0 +1,250 @@
+{%include "header.stencil" %}
+
+//
+//  Created by Adam Fowler on 2019/08/01.
+//
+//
+import Foundation
+import AWSSDKSwiftCore
+import NIO
+
+public extension S3ErrorType {
+    enum multipart : Error {
+        case noUploadId
+        case downloadEmpty(message: String)
+        case failedToOpen(file: String)
+        case failedToWrite(file: String)
+        case failedToRead(file: String)
+    }
+}
+
+public extension S3 {
+
+    /// Multipart download of a file from S3.
+    ///
+    /// - parameters:
+    ///     - input: The GetObjectRequest shape that contains the details of the object request.
+    ///     - partSize: Size of each part to be downloaded
+    ///     - outputStream: Function to be called for each downloaded part. Called with data block and file size
+    /// - returns: A future that will receive the complete file size once the multipart download has finished.
+    func multipartDownload(_ input: GetObjectRequest, partSize: Int = 5*1024*1024, on eventLoop: EventLoop, outputStream: @escaping (Data, Int64) throws -> ()) -> Future<Int64> {
```

Comment:
Minor thing, Can you update the function header comment to include the on eventLoop parameter as these are passed into the documentation